ci-operator/step-registry/gcp/provision/dns-peering-zone/gcp-provision-dns-peering-zone-commands.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+python3 --version 
+export CLOUDSDK_PYTHON=python3
+
+function run_command() {
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+}
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+dns_sa=$(yq-go r "${SHARED_DIR}/dns-peering-zone-settings.yaml" serviceAccount)
+target_project=$(yq-go r "${SHARED_DIR}/dns-peering-zone-settings.yaml" targetProject)
+target_network=$(yq-go r "${SHARED_DIR}/dns-peering-zone-settings.yaml" targetNetwork)
+
+echo "$(date -u --rfc-3339=seconds) - INFO: create consumer VPC network..."
+cmd="gcloud compute networks create ${CLUSTER_NAME}-consumer-network --subnet-mode=custom 2>/dev/null"
+run_command "${cmd}"
+
+echo "$(date -u --rfc-3339=seconds) - INFO: create DNS peering zone..."
+cmd="gcloud dns managed-zones create ${CLUSTER_NAME}-dns-peering-zone --description=\"for OCPBUGS-38719 testing\" --dns-name=${CLUSTER_NAME}.${GCP_BASE_DOMAIN}. --networks=${CLUSTER_NAME}-consumer-network --account=${dns_sa} --target-network=${target_network} --target-project=${target_project} --visibility=private"
```

Comment:
How to consume the created dns zone when running ocp installation ?

Reply:
No. Sorry, let me clarify a bit. In the scenario of [OCPBUGS-38719](https://issues.redhat.com/browse/OCPBUGS-38719), when without the bug fixing, the installation would fail quickly by telling below error, and after the bug getting fixed, the installation could move on and succeed finally. 

```
ERROR failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed provisioning resources after infrastructure ready: failed to create DNS records: failed to create record set api-int.jiwei-0914x.qe.gcp.devcluster.openshift.com.: googleapi: Error 400: Resource record sets for the zone 'jiwei-dns-peering-zone' cannot be modified because it is a Peering zone., peeringChangeDisallowed

Reply:
Okay, I guess I got the test point, sounds like we are creating a peer zone, to verify it does not break the normal OCP cluster installation.